lumicks/pylake/kymotracker/tests/test_image_sampling.py

```diff
@@ -114,3 +116,41 @@ def test_origin_warning_sample_from_image():
         ),
     ):
         tracks[0].sample_from_image(0)
+
+
+@pytest.mark.parametrize(
+    "time_idx, ref_dead_included, ref_dead_excluded",
+    [
+        ([0, 1, 2], [14.5, 24.5, 34.5], [12.5, 22.5, 32.5]),
+        ([0, 2], [14.5, 34.5], [12.5, 32.5]),
+        ([], [], []),
+    ],
+)
+def test_sample_from_channel(time_idx, ref_dead_included, ref_dead_excluded):
+    img = np.zeros((5, 5))
+    kymo = _kymo_from_array(
+        img,
+        "r",
+        line_time_seconds=1.0,
+        exposure_time_seconds=0.6,
+        start=_FIRST_TIMESTAMP + int(1e9),
+    )
+
+    data = Slice(Continuous(np.arange(100), start=_FIRST_TIMESTAMP, dt=int(1e8)))  # 10 Hz
+    kymotrack = KymoTrack(time_idx, [0, 1, 2], kymo, "red", 0)
```

Comment:
Lacks the case where the queried time indices are larger than the given track points, e.g., `time_idx = [0, 1, 2, 3]` (or even `[0, 3]`, `[2, 3]`).  I am guessing that `sample_from_channel` would fail and return an empty slice but yield no warning to the user, correct?

Or is that the scenario you test for in the `test_sample_from_channel_no_overlap` test? Note: It's not clear to me what `no_overlap` refers to in that test.

Reply:
If a time index outside the `Kymo` is required, it will raise. I don't really see how this would be possible in a real scenario though, since the `KymoTrack` is derived from the `Kymo` being sampled and the `Kymo` is private and cannot be modified.

The no overlap case is the case where the channel data has _no_ overlap with the kymotrack (which would happen if you happen to evaluate it for the wrong slice for example). It is important to handle that case, since it is not unthinkable that a user might try to match channels from the wrong file with another kymo.